source/bson-corpus/tests/binary.json

```diff
@@ -37,7 +37,8 @@
         {
             "description": "subtype 0x04",
             "canonical_bson": "1D000000057800100000000473FFD26444B34C6990E8E7D1DFC035D400",
-            "canonical_extjson": "{\"x\" : { \"$binary\" : {\"base64\" : \"c//SZESzTGmQ6OfR38A11A==\", \"subType\" : \"04\"}}}"
+            "canonical_extjson": "{\"x\" : { \"$binary\" : {\"base64\" : \"c//SZESzTGmQ6OfR38A11A==\", \"subType\" : \"04\"}}}",
+            "degenerate_extjson": "{\"x\" : { \"$uuid\" : \"73ffd264-44b3-4c69-90e8-e7d1dfc035d4\"}}"
```

Comment:
Add tests in decodeErrors for various forms of bad `$uuid values, e.g. wrong type, wrong number of characters, missing dashes, etc.

Reply:
See below, where I respectfully disagree about missing dashes.